custom/lib_portal/models/issue_book.py

```diff
@@ -18,47 +18,66 @@ def _check_due_date(self):
     book_id = fields.Many2one('product.template', string = "Book")
     book_name = fields.Char(related='book_id.name', string = "Book", 
                             store=True)
+    is_author = fields.Boolean(related='partner_id.is_author')
     author_name = fields.Char(related='book_id.author_name', 
                               string = "Author", store=True)
     from_date = fields.Date(string='Start Date',
                            default=fields.Date.context_today)
     due_date = fields.Date(string="Return Date",
                            default=_check_due_date)
     employee_id = fields.Many2one('hr.employee', string='Employee')
-    state = fields.Selection(selection=[('plan', 'Planned'),
+    state = fields.Selection([('plan', 'Planned'),
                                          ('pick', 'Picked'),
-                                         ('late', 'Late'),
-                                         ('return', 'Returned'),], 
+                                         ('return', 'Returned'),
+                                         ('late', 'Late')], 
                               string='Status',default='plan',
-                              help="The current state of your book:")
+                              help="The current state of the booking:")
+    
+    @api.model
+    def create(self, vals_list):
+        bookissue_obj = super(IssueBook, self).create(vals_list)
+        product = self.env['product.product']
+        inventory_stock = self.env['stock.quant']
+        product_id = product.search([
+            ('product_tmpl_id', '=', bookissue_obj.book_id.id)]).id
+        qty_available = inventory_stock.search([
+            ('product_id', '=', product_id),('quantity', '>=', 0)]).quantity
+        if qty_available <= 0.0 :
+            raise ValidationError('Books are not available in stock to issue!!')
+        return super().create(vals_list)
+    
+    @api.model
+    def _compute_booking_state(self):
+        print(type(dict(self._fields['state'].selection).get(self.state)))
+        return dict(self._fields['state'].selection).get(self.state)
     
     @api.constrains('due_date')
```

Comment:
this constrain should be for both fields 'due_date' and 'from_date'

Reply:
updated the code